### DISCUSSION TITLE
Access limitations for some amenities (like amenity=fuel)

### DISCUSSION THREAD

**Discussion by mnalis** [streetcomplete/StreetComplete#3066] [Ideas]
As noted in https://github.com/streetcomplete/StreetComplete/issues/3053 in some countries some amenities (like Fuel stations) limit access to some parts of them.

For example, [amenity=fuel](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dfuel) might limit access to its [compressed_air=yes](https://wiki.openstreetmap.org/wiki/Key:compressed_air) only to customers, or only to motor vehicle owners.

_(Although I'm still not quite sure how it works in such countries. Does it allow [motorcycle](https://wiki.openstreetmap.org/wiki/Key:motorcycle) to use air? How about [moped](https://wiki.openstreetmap.org/wiki/Key:moped) or [mofa](https://wiki.openstreetmap.org/wiki/Key:mofa)? How about electric mofa, which is (due to marketing) often incorrectly called e-bike (it is two wheeled vehicle that can be propelled by pedals or motor with limited power - eg. mofa, it is just that the engine is electric and not [ICE](https://en.wikipedia.org/wiki/Internal_combustion_engine)?)_

This also applies to other access rights (which might warrant separate quests); for example in my (limited) experience traveling through say Germany it is common that toilets at inter-city fuel stations are predominantly `access=customers` only, while in this part of Europe (Croatia, Bosnia&Herzegovina, Serbia...), they are virtually always unrestricted free public access `access=yes` + `fee=no`. 

This could also be  much wider (eg. access to fancy restaurant requiring specific dress code and/or only with prior reservation)

The issues to think about:

- how to mark such limitations? 
- is there a documented or de-facto standardized use of it (eg. `access:` prefix or suffix)
- are there data consumers for that tag, or should it use more generic tag?
- limiting to specific countries

**Comment by mnalis** [comment]
If the access is often limited in some countries, I think this should always be separate quest, due to [quest guidelines](https://github.com/streetcomplete/StreetComplete/wiki/Adding-new-Quests-to-StreetComplete) stating:
> :atom_symbol: Atomic quests: Per quest, only one thing should need to be answered by the user.

So if the `air_compressor` is there at fuel station, it should be separate quest from _"who has access to compressed air"_?

It should IMHO also be limited to such countries where it happens in noticeable number of quests (if it is just few cases, `create new note` is good enough and not need for separate quest). Also, such quest should then be asked not only on `amenity=fuel` nodes but also to separate `amenity=compressed_air` nodes (all of them, or only those close to `amenity=fuel`?)

Few ways it could be marked:

- Split node/area with `amenity=fuel` + `compressed_air=yes` to separate nodes: `amenity=fuel` without `compressed_air=*`, and `amenity=compressed_air` with `access=permissive`,`motor_vehicle=yes`,`bicycle=no`. While cleanest/most documented, it does not fit into SC too well (creating new nodes)

- keep one node/way, and add free format [description](https://wiki.openstreetmap.org/wiki/Key:description) tag might be added, eg. `description=compressed_air only available to motor vehicles`). This is widely supported in data consumers like OsmAnd, works in line with SC (no creating new nodes), and can/should be used either standalone or in addition with any other solutions.

- keep one node/way, and add `bicycle=no`. While less accurate, probably more widely supported.

- keep one node/way, and use `access:` / `bicycle` / `motor_vehicle` as prefix or suffix (need research which is more widely used / documented, probably suffix)? eg. `compressed_air:bicycle=no`

- keep one node/way, and use special value instead of `yes`, eg. `compressed_air=motor_vehicle`. There are few versions of `compressed_air=bicycle` so there is precedent according to [taginfo](https://taginfo.openstreetmap.org/keys/compressed_air#values)

-  use [conditional restrictions](https://wiki.openstreetmap.org/wiki/Conditional_restrictions) either as tags or as [relations](https://wiki.openstreetmap.org/wiki/Relation:restriction)?

- leaving an OSM note (simple and supported, but does not really solve the problem, only offloads to the wider community to solve it in one of ways above)

**Comment by matkoniecz** [comment]
Note that it should not be asked for all fuel station (or other objects) if only tiny minority restricts access.

It is asked for playgrounds as a noticeable part of playgrounds with access tag is actually private.

**Reply by mnalis** [reply]
I agree it should not be asked if it would be spammy (99% of the answers "not restricted"). As I've never seen it in Croatia, I assume it is country-specific thing, so this quest about who can access would probably only be enabled on a per-country basis (Germany, ...?)

Or did you @matkoniecz  had some other idea in which conditions should this quest of _"Who has `access` to `compressed_air`"_ be hidden (eg. only in presence/absence of some other tag, or nearby object... ?)

**Reply by matkoniecz** [reply]
No - I have never encountered it.

Though some petrol stations will be inaccessible to cyclists, for examples ones on motorways.

**Reply by andrewharvey** [reply]
> Though some petrol stations will be inaccessible to cyclists, for examples ones on motorways.

Most motorways near me you can and many do cycle on in the shoulder.

**Reply by mnalis** [reply]
It is IMHO not problematic if fuel station is accessible only via motorway, because if motorway is only way to get to fuel station, and it is correctly marked with `bicycle=no`, bicycles won't be routed to it. If bicycles did somehow managed to get to the fuel station (if cycling on shoulder is allowed, or pushed while walking via grass on the side of the road, or put in a car etc.) then such non-restrictive fuel station would still allow them to use air compressor
